File: server-api/internal/system/router/pprof.go
package router

import (
	handler "megin/internal/admin-api/system"
	"megin/pkg/context/router"
)

// PprofRouter 注册受后台登录权限保护的性能分析接口。
func PprofRouter(adminApiGroup *router.RouteGroup) *router.RouteGroup {
	pprof := &handler.Pprof{}
	router.GET(adminApiGroup, "/system/pprof/status", pprof.Status)
	router.POST(adminApiGroup, "/system/pprof/toggle", pprof.Toggle)
	router.POST(adminApiGroup, "/system/pprof/cpu/start", pprof.StartCPUProfile)
	router.GET(adminApiGroup, "/system/pprof/cpu/result", pprof.CPUProfileResult)
	router.GET(adminApiGroup, "/system/pprof/records", pprof.Records)
	router.GET(adminApiGroup, "/system/pprof/record", pprof.Record)
	return adminApiGroup
}

File: admin-web/src/api/pprof.js
import service from '@/utils/request'

export const getPprofStatus = () => service({
  url: '/system/pprof/status',
  method: 'get'
})

export const togglePprof = (enable) => service({
  url: '/system/pprof/toggle',
  method: 'post',
  data: { enable }
})

export const startPprofCPUProfile = (durationSeconds) => service({
  url: '/system/pprof/cpu/start',
  method: 'post',
  data: { duration_seconds: durationSeconds }
})

export const getPprofCPUProfile = () => service({
  url: '/system/pprof/cpu/result',
  method: 'get'
})

export const getPprofRecords = (params) => service({
  url: '/system/pprof/records',
  method: 'get',
  params
})

export const getPprofRecord = (id) => service({
  url: '/system/pprof/record',
  method: 'get',
  params: { id }
})
